include/lsst/meas/astrom/matchOptimisticB.h

```diff
@@ -0,0 +1,137 @@
+// -*- lsst-c++ -*-
+#if !defined(LSST_MEAS_ASTROM_MATCHOPTIMISTICB_H)
+#define LSST_MEAS_ASTROM_MATCHOPTIMISTICB_H
+
+#include <cmath>
+#include <string>
+#include <vector>
+
+#include "lsst/pex/config.h"
+#include "lsst/afw/table/Source.h"
+#include "lsst/afw/table/Match.h"
+
+namespace lsst {
+namespace meas {
+namespace astrom {
+
+    /**
+    A wrapper around a SimpleRecord that allows us to associate an pixel position
+    with a position reference star.
+
+    RecordProxy can also wrap a SourceRecord (in which case the pixel position in the proxy
+    should match that in the record).
+    */
+    struct RecordProxy {
+        PTR(lsst::afw::table::SimpleRecord) record;
+        lsst::afw::geom::Point2D position;
```

Comment:
I thought you were going to get rid of `RecordProxy` in favor of having a scratch-space centroid in the schema returned by the loader interface.  Did that turn out not to work, or have you just not gotten there yet?